pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -182,31 +197,45 @@ func NewConfigurationAccessor(
 }
 
 // list implements Accessor
-func (c *Configuration) list(ns, name string) ([]kmeta.Accessor, error) {
-	cl, err := c.configurationLister.Configurations(ns).List(labels.SelectorFromSet(labels.Set{
-		serving.RouteLabelKey: name,
-	}))
+func (c *Configuration) list(ns, routeName string, state v1.RoutingState) ([]kmeta.Accessor, error) {
+	cl, err := c.configurationLister.Configurations(ns).List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 	// Need a copy to change types in Go
-	kl := make([]kmeta.Accessor, len(cl))
-	for i, c := range cl {
-		kl[i] = c
+	kl := make([]kmeta.Accessor, 0, len(cl))
+	for _, c := range cl {
+		if _, has := getListAnnValue(c.Annotations, serving.RoutesAnnotationKey, routeName); has {
+			kl = append(kl, c)
+		}
 	}
 	return kl, err
 }
 
+func getListAnnValue(annotations map[string]string, key, value string) ([]string, bool) {
```

Comment:
this function deserves a comment.
Also given we're not saving much by doing 2 things, may be pull the contains check out
Finally it seems returning set.Strings might be a better approach here (remove from list become O(1) vs O(N)), or more precisely this code goes away.